src/state.rs

```diff
@@ -1,10 +1,35 @@
 use cosmwasm_schema::cw_serde;
-use cw_storage_plus::{Map};
+use cosmwasm_std::Coin;
+use cw_storage_plus::{Item, Map};
 
 #[cw_serde]
 pub struct Website {
     pub creator: String,
     pub source: String,
+
+    /// created_height is useful to trust the length of time a domain is up for
+    /// and also if the manager decides later to add a cooldown period from infinite before.
+    pub created_height: u64,
+
+    pub in_grace_period: bool,
 }
+pub const WEBSITES: Map<&str, Website> = Map::new("ws");
 
-pub const WEBSITES: Map<&str, Website> = Map::new("websites");
+// block height, domain names that expire at that height (moved to temp grace period)
+#[cw_serde]
+pub struct ExpireTimes {
+    pub names: Vec<String>,
+}
+pub const EXPIRE_TIMES: Map<u64, ExpireTimes> = Map::new("et");
+pub const GRACE_PERIOD: Map<u64, ExpireTimes> = Map::new("g");
```

Comment:
I do not like having both, would like to reduce this if possible. The more we can put into just the standard Website struct here is Ideal